apps/chat/src/services/credits.ts

```
@@ -5,10 +5,153 @@ export interface UserCredits {
+    const timeDiff = now.getTime() - lastResetAt.getTime()
+
+    switch (resetPeriod) {
+      case ResetPeriod.DAILY:
+        return timeDiff >= 24 * 60 * 60 * 1000 // 24 hours
+      case ResetPeriod.WEEKLY:
+        return timeDiff >= 7 * 24 * 60 * 60 * 1000 // 7 days
+      case ResetPeriod.BIWEEKLY:
+        return timeDiff >= 14 * 24 * 60 * 60 * 1000 // 14 days
+      case ResetPeriod.MONTHLY:
+        // Reset on same day of month (e.g., every 1st of month)
+        const lastResetMonth = lastResetAt.getMonth()
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)